okhttp-testing-support/src/main/java/okhttp3/ClientRuleEventListener.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import java.io.IOException
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.Proxy
+import java.util.concurrent.TimeUnit
+
+class ClientRuleEventListener(var logger: (OkHttpClient, String) -> Unit) : EventListener(),
```

Comment:
Can we combine this with our other event listener that works similarly?

Reply:
RecordingEventListener?  Good shout, I could just make that the default and pinch the events out.

Reply:
This is tough, RecordingEventListener is mutable in body of tests, doesn't have great toString etc and LoggingEventListener is in inaccessible location.  Would prefer to defer this for now.